migrate.go

```diff
@@ -753,7 +753,7 @@ func (m *Migrate) versionExists(version uint) error {
 		return err
 	}
 
-	return os.ErrNotExist
```

Comment:
Ok, now it returns a similar error so the tests do not change

Reply:
Using `os.PathError` is a much better approach!

However, this is still not a backwards compatible change since existing checks against `os.ErrNotExist` will fail in this case. e.g. existing checks will need to use `os.IsNotExists()` instead. The unit tests happen to already be using `os.IsNotExists()`.

Do you have a test case or scenario (not in the unittests where the sources are stubbed) in which a `os.PathError` will be returned?